Dockerfile

```diff
@@ -1,26 +1,50 @@
+ARG HOME=/app UID=9500 GID=9500 VERSION="base"
+ARG NODE_ENV=production
+
+# Define the base image, here we are root.
 FROM node:18 as base
 
-WORKDIR /app
+RUN touch /is-docker
+
+# Define the user and working directory, everything from this is user land
+FROM base as user
+
+ARG HOME UID GID
```

Comment:
You can specify global args above the first stage, but you still have to "pull" those args into each stage that requires them.

Args are also only available during the build, to make them available at runtime, you need to map them to ENV like that final stage does.